examples/systemcalls/systemcalls.c

```diff
@@ -83,17 +106,43 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
     // this line is to avoid a compile warning before your implementation is complete
     // and may be removed
     command[count] = command[count];
+    int status;
+    int fd = open(outputfile, O_RDWR|O_CREAT, 0644);
+    if (fd < 0) 
+    { perror("open"); 
+    exit(-1);
+    }
 
+    pid=fork();
+    if(pid==-1){
+        perror("fork");
+        return false;
+    }
+    else if(pid==0){ 
+   if (dup2(fd, 1) < 0){ 
+    perror("dup2"); 
+    exit(-1); }
+    close(fd);
+    ret=execv(command[0],command);
+    if(ret == -1){
+    perror("execv");
+    exit (EXIT_FAILURE);
+    }
+    exit (EXIT_FAILURE);
+    }
+    pid= waitpid (pid, &status, 0);
+    if(pid==-1){
+        perror("waitpid");    
+    }
```

Comment:
@Sona-Subair think about how you could avoid duplicating logic between the two `do_exec` functions.